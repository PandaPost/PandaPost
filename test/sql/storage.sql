\set ECHO none

\i test/pgxntool/setup.sql

CREATE FUNCTION pg_temp.test_value(
) RETURNS ndarray
TRANSFORM FOR TYPE ndarray
LANGUAGE plpythonu IMMUTABLE AS $body$
  import numpy as np

  return np.array([1,2,3,444]) # 444 is big enough to test endianness
$body$;
CREATE FUNCTION pg_temp.nd_as_intarray(
  i ndarray
) RETURNS int[]
TRANSFORM FOR TYPE ndarray
LANGUAGE plpythonu AS $body$
  import numpy as np

  return i.tolist()
$body$;

CREATE TEMP TABLE nd(
  nd ndarray
);

INSERT INTO nd SELECT pg_temp.test_value();

SELECT * FROM nd;

SELECT pg_temp.nd_as_intarray(nd) FROM nd;

\echo # TRANSACTION INTENTIONALLY LEFT OPEN!

-- vi: expandtab sw=2 ts=2
