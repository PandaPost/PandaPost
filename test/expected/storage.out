\set ECHO none
psql:test/pgxntool/tap_setup.sql:3: NOTICE:  42P06: schema "tap" already exists, skipping
LOCATION:  CreateSchemaCommand, schemacmds.c:115
psql:test/pgxntool/tap_setup.sql:5: NOTICE:  42710: extension "pgtap" already exists, skipping
LOCATION:  CreateExtension, extension.c:1535
\x00636e756d70792e636f72652e6d756c746961727261790a5f7265636f6e7374727563740a70310a28636e756d70790a6e6461727261790a70320a2849300a74532762270a745270330a2849310a2849340a74636e756d70790a64747970650a70340a2853276938270a49300a49310a745270350a2849330a53273c270a4e4e4e492d310a492d310a49300a74624930300a53275c7830315c7830305c7830305c7830305c7830305c7830305c7830305c7830305c7830325c7830305c7830305c7830305c7830305c7830305c7830305c7830305c7830335c7830305c7830305c7830305c7830305c7830305c7830305c7830305c7862635c7830315c7830305c7830305c7830305c7830305c7830305c783030270a74622e
{1,2,3,444}
# TRANSACTION INTENTIONALLY LEFT OPEN!
